sdk/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -134,11 +135,50 @@ public String getEndpoint() {
      * endpoint.
      *
      * @param endpoint the custom endpoint to send report to
+     * @deprecated use {@link com.bugsnag.android.Configuration#setEndpoints(String, String)}
      */
+    @Deprecated
     public void setEndpoint(String endpoint) {
         this.endpoint = endpoint;
     }
 
+    /**
+     * Set the endpoints to send data to. By default we'll send error reports to
+     * https://notify.bugsnag.com, and sessions to https://session.bugsnag.com, but you can
+     * override this if you are using Bugsnag Enterprise to point to your own Bugsnag endpoint.
+     *
+     * Please note that it is recommended that you set both endpoints. If the notify endpoint is
+     * missing, an exception will be thrown. If the session endpoint is missing, a warning will be
+     * logged and sessions will not be sent automatically.
+     *
+     * @param notify the notify endpoint
+     * @param sessions the sessions endpoint
+     *
+     * @throws IllegalArgumentException if the notify endpoint is empty or null
+     */
+    @SuppressWarnings("ConstantConditions")
+    public void setEndpoints(@NonNull String notify, @NonNull String sessions)
+        throws IllegalArgumentException {
+
+        if (TextUtils.isEmpty(notify)) {
+            throw new IllegalArgumentException("Notify endpoint cannot be empty or null.");
+        } else {
+            this.endpoint = notify;
+        }
+
+        boolean invalidSessionsEndpoint = TextUtils.isEmpty(sessions);
+
+        if (invalidSessionsEndpoint) {
+            Logger.warn("The session tracking endpoint has not been set. "
+                + "Session tracking is disabled");
+            autoCaptureSessions = false;
+            this.sessionEndpoint = null;
+        } else {
+            this.sessionEndpoint = sessions;
+        }
+        this.autoCaptureSessions = !invalidSessionsEndpoint;
```

Comment:
If this is being set in the if/else above, does it need to be set again here?  It may end up being set to `true` after it's been manually disabled.

Reply:
Both are not required, I've removed `autoCaptureSessions = false`

Reply:
There could be an issue where someone sets `autoCaptureSessions` to false, then inputs a valid sessions endpoint.  The:
```java
this.autoCaptureSessions = !invalidSessionsEndpoint;
```
would then incorrectly set the `autoCaptureSessions` back to true, overriding the user's configuration.  Is that a possibility?

Reply:
Good point - I've added an explicit test case for this scenario.